Makefile

```diff
@@ -0,0 +1,70 @@
+.POSIX:
+
+MAKEFLAGS += -r
+.DEFAULT_GOAL := default
+.DELETE_ON_ERROR:
+.SUFFIXES:
+
+VERSION ?= 2.0
+
+PWD := $(shell pwd)
+
+BUILD_ROOT ?= $(PWD)/_build
+ISO_ROOT ?= $(BUILD_ROOT)/root
+
+ALL = \
+	$(ISO_ROOT)/bootstrap.sh \
+	\
+	$(ISO_ROOT)/salt/metalk8s/containerd/configured.sls \
+	$(ISO_ROOT)/salt/metalk8s/containerd/init.sls \
+	$(ISO_ROOT)/salt/metalk8s/containerd/installed.sls \
+	\
+	$(ISO_ROOT)/salt/metalk8s/defaults.yaml \
+	\
+	$(ISO_ROOT)/salt/metalk8s/kubelet/init.sls \
+	$(ISO_ROOT)/salt/metalk8s/kubelet/installed.sls \
+	\
+	$(ISO_ROOT)/salt/metalk8s/map.jinja \
+	\
+	$(ISO_ROOT)/salt/metalk8s/repo/configured.sls \
+	$(ISO_ROOT)/salt/metalk8s/repo/init.sls \
+	$(ISO_ROOT)/salt/metalk8s/repo/offline.sls \
+	$(ISO_ROOT)/salt/metalk8s/repo/online.sls \
+	\
+	$(ISO_ROOT)/salt/metalk8s/runc/init.sls \
+	$(ISO_ROOT)/salt/metalk8s/runc/installed.sls \
+	\
+	$(ISO_ROOT)/pillar/repositories.sls \
+	$(ISO_ROOT)/pillar/top.sls \
+
+
+default: all
+.PHONY: default
+
+all: $(ALL)
+.PHONY: all
+
+$(ISO_ROOT)/bootstrap.sh: scripts/bootstrap.sh.in
+	mkdir -p $(shell dirname $@)
+	rm -f $@
+	cat $< | sed s/@VERSION@/$(VERSION)/g > $@ || rm -f $@
+	chmod a+x $@
+
+$(ISO_ROOT)/salt/%: salt/%
+	mkdir -p $(shell dirname $@)
+	rm -f $@
+	cp --archive $< $@
+
+$(ISO_ROOT)/pillar/top.sls: pillar/top.sls.in
+	mkdir -p $(shell dirname $@)
+	rm -f $@
+	cat $< | sed s/@VERSION@/$(VERSION)/g >$@ || rm -f $@
```

Comment:
Turned into `sed s/@VERSION@/$(VERSION)/g < $< > $@ || rm -f $@`

I tend to pipe in to `stdin` rather than expect a tool to take input files as an argument whenever possible, so I can be 100% certain that tool won't modify the input.

Reply:
I see, for `sed` it's safe as in-place modification need the `-i/--in-place` option.

Reply:
Yup.